logdetective/server/emoji.py

```diff
@@ -36,6 +37,23 @@ async def collect_emojis_for_mr(
     await collect_emojis_in_comments(comments, gitlab_conn)
 
 
+async def _handle_gitlab_operation(func: Callable, *args):
+    """
+    It handles errors for the specified GitLab operation.
+    After executing it in a separate thread.
+    """
+    try:
+        return await asyncio.to_thread(func, *args)
+    except gitlab.GitlabError as e:
+        log_msg = f"Error during GitLab operation {func}{args}: {e}"
+        if "Not Found" in str(e):
+            LOG.error(log_msg)
+        else:
+            LOG.info(log_msg)
+    except Exception as e:  # pylint: disable=broad-exception-caught
+        LOG.error("Unexpected error during GitLab operation %s(%s): %s", func, args, e)
+
+
```

Comment:
```suggestion
async def _handle_gitlab_operation(func: Callable, *args):
    """
    It handles errors for the specified GitLab operation.
    After executing it in a separate thread.
    """
    try:
        return await asyncio.to_thread(func, *args)
    except gitlab.GitlabError as e:
        log_msg = f"Error during GitLab operation {func}{args}: {e}"
        if "Not Found" in str(e):
            LOG.error(log_msg)
        else:
            LOG.info(log_msg)


```

At minimum, we need to drop the catch-all statement here; we want to know if something unexpected happens and have it show up in the Sentry report. Since this is happening in a background task, FastAPI will catch it anyway and log it with the appropriate backtrace.

I also don't like how broad the GitlabError catch is, though I'm not certain it'll be realistic to detect all of the probable cases; I'm going to look into narrowing that down.